### ISSUE TITLE
[Feature request]: Sorting of asset index

### ISSUE THREAD

**Issue by carlosvirreira** [Shelf-nu/shelf.nu#488]
### Is your feature request related to a problem? Please describe?

Currently my inventory is shown by create date. I would like to have a few more options on how to show my inventory on the index.

Sort by: 

- Category
- Date created (This is our default)
- A to Z

### Describe the solution you'd like

It is not a huge problem but I would prefer to see all my assets of each category bundled.

**Comment by jurrejansen** [comment]
@carlosvirreira @DonKoko Here is a scenario for sorting the index:

https://www.figma.com/file/ykvc0MhPeNLsrJ4tJsKosQ/Shelf-%7C-UX-Scenarios?type=design&node-id=1314-216141&mode=design

However, this is blocked by rework of dropdowns https://github.com/Shelf-nu/shelf.nu/issues/215

**Comment by DonKoko** [comment]
I kicked this off. The design is unfortunately not really helpful as it doesn't consider order and it has 2 options that refer to the same variable.
To do proper sorting we need to have 2 things:
- field to use for sorting
- direction of the sort
![Screenshot 2024-06-13 at 17 09 29](https://github.com/Shelf-nu/shelf.nu/assets/7840007/b976d166-f0df-4906-a5a4-9ff158c93e7f)

I want to do something similar to what supabase is doing. I did some tests in the branch but nothing final in terms of UI, was just experimenting:
![Screenshot 2024-06-13 at 17 11 16](https://github.com/Shelf-nu/shelf.nu/assets/7840007/e5593f76-8e9a-4ef2-9dba-f7f42c21f760)
